File: scripts/verify-categories.py
#!/usr/bin/env python3
"""
分类配置验证脚本（Python版本）
验证 data-service/config/categories.py 配置的正确性
"""

import sys
sys.path.insert(0, 'data-service')

from config.categories import (
    CATEGORIES,
    CATEGORY_GROUPS,
    get_all_category_codes,
    get_all_category_ids,
    get_categories_by_group,
    get_category_by_code,
    generate_ai_category_prompt
)

def main():
    print("=" * 60)
    print("分类配置验证（Python）")
    print("=" * 60)

    # 1. 基本统计
    print(f"\n✅ 总分类数: {len(CATEGORIES)}")
    print(f"✅ 分组数: {len(CATEGORY_GROUPS)}")

    # 2. 检查重复
    codes = get_all_category_codes()
    ids = get_all_category_ids()

    unique_codes = set(codes)
    unique_ids = set(ids)

    if len(codes) != len(unique_codes):
        print("❌ 发现重复的分类代码!")
        duplicates = [code for code in codes if codes.count(code) > 1]
        print(f"   重复: {set(duplicates)}")
        return False
    else:
        print("✅ 分类代码无重复")

    if len(ids) != len(unique_ids):
        print("❌ 发现重复的分类ID!")
        duplicates = [id for id in ids if ids.count(id) > 1]
        print(f"   重复: {set(duplicates)}")
        return False
    else:
        print("✅ 分类ID无重复")

    # 3. 按分组统计
    grouped = get_categories_by_group()
    print(f"\n📊 分组统计:")
    for group_key, group_info in CATEGORY_GROUPS.items():
        count = len(grouped.get(group_key, []))
        print(f"   {group_info.name}: {count}个")

    # 4. 验证分类代码列表
    print(f"\n📝 所有分类代码 ({len(codes)}个):")
    print(f"   {', '.join(codes)}")

    # 5. 测试查找函数
    print(f"\n🔍 测试查找功能:")
    test_cat = get_category_by_code('ai')
    if test_cat:
        print(f"   ✅ 找到 'ai': {test_cat.name} ({test_cat.id})")
    else:
        print(f"   ❌ 无法找到 'ai'")
        return False

    # 6. 验证AI Prompt生成
    prompt = generate_ai_category_prompt()
    if len(prompt) > 100 and "22个类别" in prompt:
        print(f"   ✅ AI Prompt生成成功 ({len(prompt)}字符)")
    else:
        print(f"   ❌ AI Prompt生成失败")
        return False

    # 7. 验证关键词
    print(f"\n🔑 关键词验证:")
    total_keywords = sum(len(cat.keywords) for cat in CATEGORIES)
    print(f"   ✅ 总关键词数: {total_keywords}")

    # 检查是否有空关键词
    empty_keywords = [cat.name for cat in CATEGORIES if not cat.keywords]
    if empty_keywords:
        print(f"   ⚠️  缺少关键词的分类: {', '.join(empty_keywords)}")

    print("\n" + "=" * 60)
    print("✅ 所有验证通过!")
    print("=" * 60)
    return True

if __name__ == '__main__':
    success = main()
    sys.exit(0 if success else 1)

File: scripts/verify-categories.ts
#!/usr/bin/env tsx
/**
 * 分类配置验证脚本（TypeScript版本）
 * 验证 src/config/categories.ts 配置的正确性
 */

import {
  CATEGORIES,
  CATEGORY_GROUPS,
  getAllCategoryCodes,
  getAllCategoryIds,
  getCategoriesByGroup,
  getCategoryByCode,
  generateAICategoryPrompt,
  generateUIFilterGroups,
} from '../src/config/categories'

function main() {
  console.log('='.repeat(60))
  console.log('分类配置验证（TypeScript）')
  console.log('='.repeat(60))

  let hasError = false

  // 1. 基本统计
  console.log(`\n✅ 总分类数: ${CATEGORIES.length}`)
  console.log(`✅ 分组数: ${Object.keys(CATEGORY_GROUPS).length}`)

  // 2. 检查重复
  const codes = getAllCategoryCodes()
  const ids = getAllCategoryIds()

  const uniqueCodes = new Set(codes)
  const uniqueIds = new Set(ids)

  if (codes.length !== uniqueCodes.size) {
    console.error('❌ 发现重复的分类代码!')
    const duplicates = codes.filter((code, index) => codes.indexOf(code) !== index)
    console.error(`   重复: ${[...new Set(duplicates)].join(', ')}`)
    hasError = true
  } else {
    console.log('✅ 分类代码无重复')
  }

  if (ids.length !== uniqueIds.size) {
    console.error('❌ 发现重复的分类ID!')
    const duplicates = ids.filter((id, index) => ids.indexOf(id) !== index)
    console.error(`   重复: ${[...new Set(duplicates)].join(', ')}`)
    hasError = true
  } else {
    console.log('✅ 分类ID无重复')
  }

  // 3. 按分组统计
  const grouped = getCategoriesByGroup()
  console.log(`\n📊 分组统计:`)
  for (const [groupKey, groupInfo] of Object.entries(CATEGORY_GROUPS)) {
    const count = grouped[groupKey as keyof typeof grouped]?.length || 0
    console.log(`   ${groupInfo.name}: ${count}个`)
  }

  // 4. 验证分类代码列表
  console.log(`\n📝 所有分类代码 (${codes.length}个):`)
  console.log(`   ${codes.join(', ')}`)

  // 5. 测试查找函数
  console.log(`\n🔍 测试查找功能:`)
  const testCat = getCategoryByCode('ai')
  if (testCat) {
    console.log(`   ✅ 找到 'ai': ${testCat.name} (${testCat.id})`)
  } else {
    console.error(`   ❌ 无法找到 'ai'`)
    hasError = true
  }

  // 6. 验证AI Prompt生成
  const prompt = generateAICategoryPrompt()
  if (prompt.length > 100 && prompt.includes('22个类别')) {
    console.log(`   ✅ AI Prompt生成成功 (${prompt.length}字符)`)
  } else {
    console.error(`   ❌ AI Prompt生成失败`)
    hasError = true
  }

  // 7. 验证UI筛选器生成
  const filterGroups = generateUIFilterGroups()
  if (filterGroups.length > 0) {
    console.log(`   ✅ UI筛选器生成成功 (${filterGroups.length}个分组)`)
  } else {
    console.error(`   ❌ UI筛选器生成失败`)
    hasError = true
  }

  // 8. 验证关键词
  console.log(`\n🔑 关键词验证:`)
  const totalKeywords = CATEGORIES.reduce((sum, cat) => sum + cat.keywords.length, 0)
  console.log(`   ✅ 总关键词数: ${totalKeywords}`)

  // 检查是否有空关键词
  const emptyKeywords = CATEGORIES.filter(cat => cat.keywords.length === 0)
  if (emptyKeywords.length > 0) {
    console.warn(`   ⚠️  缺少关键词的分类: ${emptyKeywords.map(c => c.name).join(', ')}`)
  }

  // 9. 验证命名规范
  console.log(`\n📏 命名规范检查:`)
  const invalidCodes = CATEGORIES.filter(cat => !/^[a-z_]+$/.test(cat.code))
  const invalidIds = CATEGORIES.filter(cat => !/^cat_[a-z_]+$/.test(cat.id))

  if (invalidCodes.length > 0) {
    console.error(`   ❌ 不符合规范的code: ${invalidCodes.map(c => c.code).join(', ')}`)
    hasError = true
  } else {
    console.log(`   ✅ 所有code符合命名规范`)
  }

  if (invalidIds.length > 0) {
    console.error(`   ❌ 不符合规范的id: ${invalidIds.map(c => c.id).join(', ')}`)
    hasError = true
  } else {
    console.log(`   ✅ 所有id符合命名规范`)
  }

  // 10. 验证code和id的对应关系
  console.log(`\n🔗 验证code和id对应:`)
  const mismatch = CATEGORIES.filter(cat => cat.id !== `cat_${cat.code}`)
  if (mismatch.length > 0) {
    console.error(`   ❌ code和id不对应:`)
    mismatch.forEach(cat => {
      console.error(`      ${cat.code} -> ${cat.id} (预期: cat_${cat.code})`)
    })
    hasError = true
  } else {
    console.log(`   ✅ 所有code和id正确对应`)
  }

  console.log('\n' + '='.repeat(60))
  if (hasError) {
    console.error('❌ 验证失败，请修复上述错误')
    console.log('='.repeat(60))
    process.exit(1)
  } else {
    console.log('✅ 所有验证通过!')
    console.log('='.repeat(60))
    process.exit(0)
  }
}

main()
